solr/core/src/java/org/apache/solr/request/json/RequestUtil.java

```diff
@@ -190,6 +193,9 @@ public static void processParams(SolrRequestHandler handler, SolrQueryRequest re
       }
     }
 
+    List<SearchComponent> components = ((SearchHandler) handler).getComponents();
+    Set<String> pluginNames = components.stream().map(c -> c.getJsonKey()).filter(c -> c != null).collect(Collectors.toSet());
```

Comment:
nit: `pluginJsonKeys` instead (I think the naming here was from the old way you did it)